lib/OrdinaryDiffEqFIRK/src/firk_caches.jl

```diff
@@ -518,7 +519,11 @@ function alg_cache(alg::AdaptiveRadau, u, rate_prototype, ::Type{uEltypeNoUnits}
     num_stages = min
 
     tabs = [RadauIIATableau5(uToltype, constvalue(tTypeNoUnits)), RadauIIATableau9(uToltype, constvalue(tTypeNoUnits)), RadauIIATableau13(uToltype, constvalue(tTypeNoUnits))]
-    i = 9
+    if (min == 3 || min == 5 || min == 7)
+        i = 9
+    else
+        i = min
+    end
```

Comment:
`i = max(min_stages, 9)` (although this would require you to rename `min` and `max` to `min_stages`  and `max` to `max_stages` which you should do anyway.